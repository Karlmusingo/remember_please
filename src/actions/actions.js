import axios from "axios";

export const fetchEvents = () => async dispatch => {
  const { data: payload } = await axios.get(
    `${process.env.BACKEND_URL}/events/`
  );

  return dispatch({
    type: "FETCH_EVENTS",
    payload
  });
};
export const addEvent = event => async () => {
  try {
    const day = event.date.split("/")[0];
    const month = event.date.split("/")[1];
    const year = event.date.split("/")[2] || 2019;
    const target = event.name;
    const phonenumber = event.whatsappNumber;
    const data = {
      day,
      month,
      year,
      target,
      message: event.message,
      phonenumber,
      country: "Rwanda",
      notificationTime: 7,
      type: "Birthday"
    };

    const res = await axios.post(`${process.env.BACKEND_URL}/events/`, data);

    return res.data;
  } catch (error) {
    return { error: "Oops" };
  }
};
