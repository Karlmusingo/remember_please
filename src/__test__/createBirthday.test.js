import React from "react";
import { shallow, mount } from "enzyme";
import mockAxios from "axios";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import promiseMiddleware from "redux-promise-middleware";
import { fetchEvents, addEvent } from "../actions/actions";
import CreateBirthday from "../components/createBirthday";
import { jsxText } from "@babel/types";

const mockStore = configureMockStore([thunk]);

describe("Create birthday component", () => {
  let store;
  const event = {
    date: "12/04/2019",
    name: "karl",
    message: "message",
    whatsappNumber: "+23453456456"
  };
  beforeEach(() => {
    store = mockStore({
      events: {}
    });
  });

  test("Should match the snapshot", async () => {
    mockAxios.post.mockImplementationOnce(() =>
      Promise.resolve({
        data: { id: 1, name: "karl" }
      })
    );
    // await store.dispatch(fetchEvents(events));
    // const actions = store.getActions();

    // expect.assertions(3);
    // expect(actions[0].type).toEqual("FETCH_EVENTS");
    // expect(actions[0].payload).toEqual("mocked");

    expect(1).toBe(1);
  });

  test("Should match the snapshot", async () => {
    const wrapper = mount(<CreateBirthday state={event} store={store} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Should call the push history", async () => {
    const handleSubmit = jest.fn();
    const historyMock = { push: jest.fn() };
    const wrapper = mount(
      <CreateBirthday store={store} history={historyMock} />
    );
    wrapper.find(".btn").simulate("click");
    expect(1).toBe(1);
  });
});
