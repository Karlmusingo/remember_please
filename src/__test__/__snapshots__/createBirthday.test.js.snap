// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create birthday component Should match the snapshot 1`] = `
<Connect(CreateBirthday)
  state={
    Object {
      "date": "12/04/2019",
      "message": "message",
      "name": "karl",
      "whatsappNumber": "+23453456456",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <CreateBirthday
    onAddEvent={[Function]}
    state={
      Object {
        "date": "12/04/2019",
        "message": "message",
        "name": "karl",
        "whatsappNumber": "+23453456456",
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <div>
      <Navbar
        title="New Birthday"
      >
        <div
          className="container"
          style={
            Object {
              "alignText": "center",
              "background": "#ffc3b8",
              "color": "#fff",
            }
          }
        >
          <div
            className="row"
            style={
              Object {
                "display": "flex",
                "justifyContent": "space-between",
                "padding": "10px 25px 0px 20px",
              }
            }
          >
            <div
              style={
                Object {
                  "color": "#fff",
                }
              }
            >
              <p
                id="btn-cancel"
                onClick={[Function]}
              >
                Cancel
              </p>
            </div>
            <div>
              <span>
                 
                New Birthday
                 
              </span>
            </div>
          </div>
        </div>
      </Navbar>
      <div
        className="container"
        style={
          Object {
            "marginBottom": "30px",
            "marginTop": "20px",
          }
        }
      >
        <form
          onSubmit={[Function]}
        >
          <Input
            label="Name"
            name="name"
            onChange={[Function]}
            placeholder="Name"
          >
            <div
              className="form-group"
            >
              <label
                style={
                  Object {
                    "marginBottom": "-18px",
                    "marginLeft": "10px",
                  }
                }
              >
                Name
              </label>
              <input
                className="form-control"
                id="name"
                name="name"
                onChange={[Function]}
                placeholder="Name"
                style={
                  Object {
                    "border": "none",
                    "borderBottom": "2px solid #f5f5f5",
                  }
                }
                type="text"
              />
            </div>
          </Input>
          <Input
            label="Date"
            name="date"
            onChange={[Function]}
            placeholder="30/12"
          >
            <div
              className="form-group"
            >
              <label
                style={
                  Object {
                    "marginBottom": "-18px",
                    "marginLeft": "10px",
                  }
                }
              >
                Date
              </label>
              <input
                className="form-control"
                id="date"
                name="date"
                onChange={[Function]}
                placeholder="30/12"
                style={
                  Object {
                    "border": "none",
                    "borderBottom": "2px solid #f5f5f5",
                  }
                }
                type="text"
              />
            </div>
          </Input>
          <Input
            label="Whatsapp Number"
            name="whatsappNumber"
            onChange={[Function]}
            placeholder="+250781959044"
          >
            <div
              className="form-group"
            >
              <label
                style={
                  Object {
                    "marginBottom": "-18px",
                    "marginLeft": "10px",
                  }
                }
              >
                Whatsapp Number
              </label>
              <input
                className="form-control"
                id="whatsappNumber"
                name="whatsappNumber"
                onChange={[Function]}
                placeholder="+250781959044"
                style={
                  Object {
                    "border": "none",
                    "borderBottom": "2px solid #f5f5f5",
                  }
                }
                type="text"
              />
            </div>
          </Input>
          <Hr>
            <hr
              style={
                Object {
                  "backgroundColor": "#fff",
                  "color": "#f5f5f5",
                  "margin": "1px",
                }
              }
            />
          </Hr>
          <p
            className="text"
          >
            Compose your Birthday message here:
          </p>
          <Hr>
            <hr
              style={
                Object {
                  "backgroundColor": "#fff",
                  "color": "#f5f5f5",
                  "margin": "1px",
                }
              }
            />
          </Hr>
          <TextAreA
            height="200px"
            onChange={[Function]}
          >
            <textarea
              className="form-control"
              name="message"
              onChange={[Function]}
              placeholder="My message"
              style={
                Object {
                  "border": "none",
                  "borderBottom": "2px solid #f5f5f5",
                  "height": "200px",
                }
              }
            />
          </TextAreA>
          <div
            style={
              Object {
                "display": "flex",
                "justifyContent": "space-around",
                "marginTop": "15px",
              }
            }
          >
            <button
              className="btn col-5 btn-lg btn-success"
              onClick={[Function]}
            >
              Done
            </button>
          </div>
        </form>
      </div>
    </div>
  </CreateBirthday>
</Connect(CreateBirthday)>
`;
